File: software/examples/ruby/example_callback.rb
#!/usr/bin/env ruby
# -*- ruby encoding: utf-8 -*-

require 'tinkerforge/ip_connection'
require 'tinkerforge/bricklet_rgb_led_button'

include Tinkerforge

HOST = 'localhost'
PORT = 4223
UID = 'XYZ' # Change XYZ to the UID of your RGB LED Button Bricklet

ipcon = IPConnection.new # Create IP connection
rlb = BrickletRGBLEDButton.new UID, ipcon # Create device object

ipcon.connect HOST, PORT # Connect to brickd
# Don't use device before ipcon is connected

# Register button state changed callback
rlb.register_callback(BrickletRGBLEDButton::CALLBACK_BUTTON_STATE_CHANGED) do |state|
  if state == BrickletRGBLEDButton::BUTTON_STATE_PRESSED
    puts "State: Pressed"
  elsif state == BrickletRGBLEDButton::BUTTON_STATE_RELEASED
    puts "State: Released"
  end
end

puts 'Press key to exit'
$stdin.gets
ipcon.disconnect

File: software/examples/csharp/ExampleCallback.cs
using System;
using Tinkerforge;

class Example
{
	private static string HOST = "localhost";
	private static int PORT = 4223;
	private static string UID = "XYZ"; // Change XYZ to the UID of your RGB LED Button Bricklet

	// Callback function for button state changed callback
	static void ButtonStateChangedCB(BrickletRGBLEDButton sender, byte state)
	{
		if(state == BrickletRGBLEDButton.BUTTON_STATE_PRESSED)
		{
			Console.WriteLine("State: Pressed");
		}
		else if(state == BrickletRGBLEDButton.BUTTON_STATE_RELEASED)
		{
			Console.WriteLine("State: Released");
		}
	}

	static void Main()
	{
		IPConnection ipcon = new IPConnection(); // Create IP connection
		BrickletRGBLEDButton rlb = new BrickletRGBLEDButton(UID, ipcon); // Create device object

		ipcon.Connect(HOST, PORT); // Connect to brickd
		// Don't use device before ipcon is connected

		// Register button state changed callback to function ButtonStateChangedCB
		rlb.ButtonStateChangedCallback += ButtonStateChangedCB;

		Console.WriteLine("Press enter to exit");
		Console.ReadLine();
		ipcon.Disconnect();
	}
}
